www/src/components/site-metadata.js

```diff
@@ -21,28 +21,43 @@ const SiteMetadata = ({ pathname }) => {
     }
   `)
 
-  return (
-    <Helmet defaultTitle={title} titleTemplate={`%s | ${title}`}>
-      <html lang="en" />
-      <link rel="canonical" href={`${siteUrl}${pathname}`} />
-      <meta name="docsearch:version" content="2.0" />
-      <meta
-        name="viewport"
-        content="width=device-width,initial-scale=1,shrink-to-fit=no,viewport-fit=cover"
-      />
-
-      <meta property="og:url" content={siteUrl} />
-      <meta property="og:type" content="website" />
-      <meta property="og:locale" content="en" />
-      <meta property="og:site_name" content={title} />
-      <meta property="og:image" content={`${siteUrl}${gatsbyIcon}`} />
-      <meta property="og:image:width" content="512" />
-      <meta property="og:image:height" content="512" />
+  const {
+    metaImage = `${siteUrl}${gatsbyIcon}`,
+    metaTitle = title,
+    metaDescription = `${siteUrl}`,
+  } = props
 
-      <meta name="twitter:card" content="summary" />
-      <meta name="twitter:site" content={twitter} />
+  const generateMetaTags = (type, content) => (
+    <Helmet>
+      <meta property={`og:${type}`} content={content} />
+      <meta name={`twitter:${type}`} content={content} />
     </Helmet>
   )
+
+  return (
+    <React.Fragment>
+      <Helmet>
+        <html lang="en" />
+        <link rel="canonical" href={`${siteUrl}${props.pathname}`} />
+        <meta name="docsearch:version" content="2.0" />
+        <meta
+          name="viewport"
+          content="width=device-width,initial-scale=1,shrink-to-fit=no,viewport-fit=cover"
+        />
+        <meta property="og:url" content={siteUrl} />
+        <meta property="og:type" content="website" />
+        <meta property="og:locale" content="en" />
+        <meta property="og:site_name" content={metaTitle} />
+        <meta property="og:image:width" content="512" />
+        <meta property="og:image:height" content="512" />
+        <meta name="twitter:card" content="summary_large_image" />
+        <meta name="twitter:site" content={twitter} />
+      </Helmet>
+      {generateMetaTags(`description`, metaDescription)}
```

Comment:
Maybe just iterate over these as additional `meta` tags within the above Helmet? (e.g. line 54 currently)

Reply:
Are you referring to lines 53 and 54 as the additional `meta` tags? I'm not sure I follow what you are suggesting, could you please explain?

Reply:
Something like this (inside of the existing Helmet tags)

```jsx
{
  []
    .concat(generateMetaTags(`description`, metaDescription))
    .concat(generateMetaTags(`whatever`, metaDescription))
    .concat(generateMetaTags(`other`, metaDescription))
    .map(({ property, description }, index) => <meta property={property} description={description} key={`${index-${description`} />)
}
```

You'll need to tweak what generateMetaTags returns (or just add a key property and return the element when mapping).

Reply:
Gotcha, I see what you are saying. I didn't realize that most of the `Twitter` specific meta tags are unnecessary as long as there are the `og` counter parts.

I liked the idea you had of just passing a `meta` prop...so I implemented that to `Layout` and subsequently `SiteMetaData`. It will now accept an array of objects: 
```
<Layout
  meta={[
    { property: 'description', content: 'test' },
    { property: 'image', content: 'imagePath' },
    { property: 'title', content: 'This is the title' },
   ]}
/>
```
Which now fits nicely inside the one `Helmet` wrapper in `SiteMetaData`

```
<Helmet>
  <html lang="en" />
  <link rel="canonical" href={`${siteUrl}${pathname}`} />
  <meta name="docsearch:version" content="2.0" />
  <meta
    name="viewport"
    content="width=device-width,initial-scale=1,shrink-to-fit=no,viewport-fit=cover"
  />
  <meta property="og:url" content={siteUrl} />
  <meta property="og:type" content="website" />
  <meta property="og:locale" content="en" />
  <meta property="og:site_name" content={title} />
  <meta property="og:image:width" content="512" />
  <meta property="og:image:height" content="512" />
  <meta name="twitter:card" content="summary_large_image" />
  <meta name="twitter:site" content={twitter} />
{
  meta.map(({ property, content }, index) => {
    return (
      <meta property={`og:${property}`} content={content} key={`${index}-${property}`}/>
    )
  })
}
</Helmet>
```

Thoughts?

Reply:
@rlynn523 I don't think the iteration is helpful now! I'd just manually create those as JSX, e.g.

```jsx
<Helmet>
  <html lang="en" />
  <link rel="canonical" href={`${siteUrl}${pathname}`} />
  <meta name="docsearch:version" content="2.0" />
  <meta
    name="viewport"
    content="width=device-width,initial-scale=1,shrink-to-fit=no,viewport-fit=cover"
  />
  <meta property="og:url" content={siteUrl} />
  <meta property="og:type" content="website" />
  <meta property="og:locale" content="en" />
  <meta property="og:site_name" content={title} />
  <meta property="og:image:width" content="512" />
  <meta property="og:image:height" content="512" />
  <meta name="twitter:card" content="summary_large_image" />
  <meta name="twitter:site" content={twitter} />
  <meta property=og:description" content={props.description || title} />
  {/* and the rest of them */}
```

Reply:
I thought you were suggesting to use an iteration?

Reply:
> I thought you were suggesting to use an iteration?
> 
> I can just put in JSX tags. I did notice that when navigating from the root of the Gatsby site to, for example, a showcase site, the meta props don't get updated, only when you hit the page directly. I haven't looked into to the rendering of the `Layout` component. We could remove those tags from `SiteMetaData`, make a `SocialMetaTags` component that can just be imported to each page that wants to add Twitter cards.

Hey @DSchau , just wanted to check in and see if there was any other feedback/updates that should be made to this in order to have it merged. Thanks!